remote-continuous-query/src/main/java/org/infinispan/tutorial/simple/remote/query/InfinispanRemoteContinuousQuery.java

```diff
@@ -0,0 +1,151 @@
+         "anistor",
+         "karesti",
+         "ttarrant",
+         "belen_esteban",
+         "dberindei",
+         "galderz",
+         "wburns",
+         "pruivo",
+         "oliveira",
+         "vrigamonti");
+
+   // HASHTAGS will be used to create random posts
+   private static final List<String> HASHTAGS = Arrays.asList(
+         "love",
+         "instagood",
+         "photooftheday",
+         "fashion",
+         "beautiful",
+         "happy",
+         "cute",
+         "tbt",
+         "like4like",
+         "followme",
+         "infinispan");
+
+   private static final String CACHE_NAME = "instaposts";
+
+   public static void main(String[] args) throws Exception {
+      // Create a configuration for a locally-running server
+      ConfigurationBuilder builder = new ConfigurationBuilder();
+      builder.addServer().host("127.0.0.1")
+            .port(ConfigurationProperties.DEFAULT_HOTROD_PORT)
+            .marshaller(ProtoStreamMarshaller.class); // You need to specify the marshaller
+
+      // Connect to the server
+      RemoteCacheManager client = new RemoteCacheManager(builder.build());
+
+      // Get the cache, create it if needed with the default configuration
+      RemoteCache<String, InstaPost> instaPostsCache = client.administration().getOrCreateCache(CACHE_NAME, "default");
+
+      // Create and add the marshalling configuration for Person pojo class. Note Person is an annotated POJO
+      addInstapostsSchema(client);
+
+      // Get a query factory from the cache
+      QueryFactory queryFactory = Search.getQueryFactory(instaPostsCache);
+
+      // Create a query with lastName parameter
+      Query query = queryFactory.create("FROM tutorial.InstaPost p where p.user = :userName");
+
+      // Set the parameter value
+      query.setParameter("userName", "belen_esteban");
+
+      // Create the continuous query
+      ContinuousQuery<String, InstaPost> continuousQuery = Search.getContinuousQuery(instaPostsCache);
+
+      // Create the continuous query listener.
+      ContinuousQueryListener<String, InstaPost> listener =
+            new ContinuousQueryListener<String, InstaPost>() {
+               // This method will be executed every time new items that correspond with the query arrive
+               @Override
+               public void resultJoining(String id, InstaPost post) {
+                  System.out.println(String.format("@%s has posted again! Hashtag: #%s", post.user, post.hashtag));
+               }
+            };
+      // Clear all the listeners
+      continuousQuery.removeAllListeners();
+
+      // And the listener corresponding the query to the continuous query
+      continuousQuery.addContinuousQueryListener(query, listener);
+
+      // Add 1000 random posts
+      for (int i = 0; i < 1000; i++) {
+         // Add a post
+         addRandomPost(instaPostsCache);
+
+         // Await a little to see results
+         Thread.sleep(10);
+      }
+
+      // Clear the cache
```

Comment:
This actually removes the cache instead of just clearing it  :)

Reply:
changing this the comment